app/src/main/java/com/zeapo/pwdstore/ui/dialogs/XkPasswordGeneratorDialogFragment.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+package com.zeapo.pwdstore.ui.dialogs
+
+import android.app.Activity
+import android.app.Dialog
+import android.content.Context
+import android.content.SharedPreferences
+import android.graphics.Typeface
+import android.os.Bundle
+import android.util.Log
+import android.widget.CheckBox
+import android.widget.EditText
+import android.widget.Spinner
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.AppCompatEditText
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.zeapo.pwdstore.R
+import com.zeapo.pwdstore.pwgen.PasswordGenerator.PasswordGeneratorExeption
+import com.zeapo.pwdstore.pwgenxkpwd.CapType
+import com.zeapo.pwdstore.pwgenxkpwd.PasswordBuilder
+
+/** A placeholder fragment containing a simple view.  */
+class XkPasswordGeneratorDialogFragment : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = MaterialAlertDialogBuilder(requireContext())
+        val callingActivity: Activity = requireActivity()
+        val inflater = callingActivity.layoutInflater
+        val view = inflater.inflate(R.layout.fragment_xkpwgen, null)
+
+        val monoTypeface = Typeface.createFromAsset(callingActivity.assets, "fonts/sourcecodepro.ttf")
+
+        builder.setView(view)
+
+        prefs = callingActivity.getSharedPreferences("PasswordGenerator", Context.MODE_PRIVATE)
+
+        cbNumbers = view.findViewById<CheckBox>(R.id.xknumerals)
+        cbNumbers.isChecked = prefs.getBoolean(PREF_KEY_USE_NUMERALS, false)
+
+        spinnerNumbersCount = view.findViewById<Spinner>(R.id.xk_numbers_count)
+
+        val storedNumbersCount = prefs.getInt(PREF_KEY_NUMBERS_COUNT, 0)
+        spinnerNumbersCount.setSelection(storedNumbersCount)
+
+        cbSymbols = view.findViewById<CheckBox>(R.id.xksymbols)
+        cbSymbols.isChecked = false != prefs.getBoolean(PREF_KEY_USE_SYMBOLS, false)
+        spinnerSymbolsCount = view.findViewById<Spinner>(R.id.xk_symbols_count)
+        val symbolsCount = prefs.getInt(PREF_KEY_SYMBOLS_COUNT, 0)
+        spinnerSymbolsCount.setSelection(symbolsCount)
+
+        val previousStoredCapStyle: String = try {
+            prefs.getString(PREF_KEY_CAPITALS_STYLE, DEFAULT_CAPS_STYLE)
+        } catch (e: Exception) {
+            Log.e("xkpw", e.message)
```

Comment:
```suggestion
            Timber.tag("xkpw").e(e.message)
```

Use `timber.log.Timber` for logging.